rh_region.go

```diff
@@ -0,0 +1,61 @@
+package sdk
+
+import (
+	json2 "encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+type Region struct {
+	URL string
+	AWS []string
+	GCP []string
+}
+
+func GetRhRegions(ocmServiceUrl string) (map[string]Region, error) {
+	var regions map[string]Region
+	url, err := DetermineRegionDiscoveryUrl(ocmServiceUrl)
+	if err != nil {
+		return regions, fmt.Errorf("Can't determine region discovery URL: %s\n", err)
+	}
+	// Adding nolint here in order to prevent linter from failing due to variable http get
+	resp, err := http.Get(url) //nolint
+	if err != nil {
+		return regions, fmt.Errorf("Can't retrieve shards: %s", err)
+	}
+	err = json2.NewDecoder(resp.Body).Decode(&regions)
+	if err != nil {
+		return regions, fmt.Errorf("Can't decode shards: %s", err)
+	}
+	return regions, nil
+}
+
+func GetRhRegion(ocmServiceUrl string, regionName string) (Region, error) {
+	regions, err := GetRhRegions(ocmServiceUrl)
+	if err != nil {
+		return Region{}, err
+	}
+	regionName = fmt.Sprintf("rh-%s", regionName)
```

Comment:
the "rh-" prefix assumption is still here and in OCM-5281, again, dont force a key-prefix naming convention, just look for the `regionName` in the map as it is passed to the function. 

As its currently implemented, this function would find "singapore" in the map `{ "rh-singapre": foo }` but it would not find "rh-singapore" (because it would look for "rh-rh-singapre") which is confusing. We also dont want to dynamically add the "rh-" prefix because then neither "singapore" nor "rh-singapore" would work for the map  `{ "singapre": foo }` because now the map data is "wrong". 

Smart keys are rarely worth it and should only be used if you have a very hard key format requirement that you need to enforce.

Reply:
Changed to logic to check for `contains` of a region (we can change this as and when the shards json is finalized)